### PR TITLE
Add constructor for DioExceptionType.badCertificate

### PR DESCRIPTION
**Pull request by hgraceb**
- Improve some document comments.
- Change the parameter position of `DioException`'s constructors to make it most readable.
- Add constructor for `DioExceptionType.badCertificate`. There is also one reference in the plugin `http2_adapter`. But I don't know if changing that with an upgraded version is necessary.
https://github.com/cfug/dio/blob/1c2843a2fab1c3795272df1e81502616761e192e/plugins/http2_adapter/lib/src/connection_manager_imp.dart#L110-L115

### New Pull Request Checklist

- [x] I have read the [Documentation](https://pub.dev/documentation/dio/latest/)
- [x] I have searched for a similar pull request in the [project](https://github.com/cfug/dio/pulls) and found none
- [x] I have updated this branch with the latest `main` branch to avoid conflicts (via merge from master or rebase)
- [ ] I have added the required tests to prove the fix/feature I'm adding
- [ ] I have updated the documentation (if necessary)
- [ ] I have run the tests without failures
- [x] I have updated the `CHANGELOG.md` in the corresponding package

### CODE REVIEW

File: dio/lib/src/dio_exception.dart
Comment (ueman) [inline dio/lib/src/dio_exception.dart:154]: Isn't it just an "invalid" certificate where invalid is defined either by the app or the environment the app is running in? "approved" seems somewhat misleading if that's the case

Comment (hgraceb) [reply dio/lib/src/dio_exception.dart:154]: Per,

https://github.com/cfug/dio/blob/1c2843a2fab1c3795272df1e81502616761e192e/dio/lib/src/adapters/io_adapter.dart#L51-L57

https://github.com/cfug/dio/blob/1c2843a2fab1c3795272df1e81502616761e192e/dio/lib/src/adapters/io_adapter.dart#L183-L199

The message means that the user does not approve the certificate of the response by themselves.

Comment (ueman) [reply dio/lib/src/dio_exception.dart:154]: Got it, thanks!

**Comment by ueman** [comment]
> There is also one reference in the plugin http2_adapter. But I don't know if changing that with an upgraded version is necessary.

I'm fine with adding a comment there or creating an issue to fix it later. That way we don't forget about it.

**Comment by hgraceb** [comment]
> I'm fine with adding a comment there or creating an issue to fix it later. That way we don't forget about it.

I updated and added a TODO comment to do that. What do you think?

```diff
+// TODO(EVERYONE): Replace with DioException.badCertificate once upgrade dependencies Dio above 5.4.2.
 throw DioException(
   requestOptions: options,
   type: DioExceptionType.badCertificate,
   error: certificate,
   message: 'The certificate of the response is not approved.',
 );
```

**Review by ueman — APPROVED** [review]

**Comment by kuhnroyal** [comment]
@hgraceb Our pipeline should be fine now. Can you please rebase or merge main?

**Comment by hgraceb** [comment]
@kuhnroyal Nice works!

**Pull request merged**
